src/components/dashboard/sidebar/SubscriptionSummaryCard.jsx

```diff
@@ -16,43 +19,70 @@ import {
   SUBSCRIPTION_WARNING_BADGE_VARIANT,
 } from './data/constants';
 
-const SubscriptionSummaryCard = ({ subscriptionPlan, className }) => {
-  const { daysUntilExpiration, expirationDate } = subscriptionPlan;
-  const renderCardTitle = () => {
-    // Active Subscription
-    let statusBadgeLabel = SUBSCRIPTION_ACTIVE_BADGE_LABEL;
-    let statusBadgeVariant = SUBSCRIPTION_ACTIVE_BADGE_VARIANT;
-    if (daysUntilExpiration <= SUBSCRIPTION_EXPIRED) {
-      // Subscription has expired
-      statusBadgeLabel = SUBSCRIPTION_EXPIRED_BADGE_LABEL;
-      statusBadgeVariant = SUBSCRIPTION_EXPIRED_BADGE_VARIANT;
-    } else if (daysUntilExpiration <= SUBSCRIPTION_DAYS_REMAINING_SEVERE) {
-      // Expiration is approaching
-      statusBadgeLabel = SUBSCRIPTION_WARNING_BADGE_LABEL;
-      statusBadgeVariant = SUBSCRIPTION_WARNING_BADGE_VARIANT;
+const SubscriptionSummaryCard = ({ subscriptionPlan, licenseRequest, className }) => {
+  const badgeVariantAndLabel = useMemo(() => {
+    if (subscriptionPlan) {
+      if (subscriptionPlan.daysUntilExpiration <= SUBSCRIPTION_EXPIRED) {
+        // Subscription has expired
+        return ({
+          variant: SUBSCRIPTION_EXPIRED_BADGE_VARIANT,
+          label: SUBSCRIPTION_EXPIRED_BADGE_LABEL,
+        });
+      } if (subscriptionPlan.daysUntilExpiration <= SUBSCRIPTION_DAYS_REMAINING_SEVERE) {
+        // Expiration is approaching
+        return ({
+          variant: SUBSCRIPTION_WARNING_BADGE_VARIANT,
+          label: SUBSCRIPTION_WARNING_BADGE_LABEL,
+        });
+      }
+
+      return ({
+        variant: SUBSCRIPTION_ACTIVE_BADGE_VARIANT,
+        label: SUBSCRIPTION_ACTIVE_BADGE_LABEL,
+      });
+    }
+
+    if (licenseRequest) {
```

Comment:
NIT: I might write this as an else if just to ensure it never gets called if there is a subscription plan. It's OK as it's written now, but the else if would leave it in a better defensive state if further changes were made in the future in this area.

Reply:
it should never get to this branch if there is a subscription plan, and es-lint would consider it a `Unnecessary 'else' after 'return'.